File: benchmarks/tsb/bench_dataframe_transform_rows.ts
/**
 * Benchmark: dataFrameTransformRows on 10k-row DataFrame
 */
import { DataFrame, dataFrameTransformRows } from "../../src/index.js";

const ROWS = 10_000;
const WARMUP = 3;
const ITERATIONS = 10;
const a = Array.from({ length: ROWS }, (_, i) => i * 1.0);
const b = Array.from({ length: ROWS }, (_, i) => i * 2.0);
const df = DataFrame.fromColumns({ a, b });

for (let i = 0; i < WARMUP; i++)
  dataFrameTransformRows(df, (row) => ({ a: (row.a as number) * 2, b: (row.b as number) + 1 }));
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++)
  dataFrameTransformRows(df, (row) => ({ a: (row.a as number) * 2, b: (row.b as number) + 1 }));
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "dataframe_transform_rows",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_dataframe_transform_rows.py
"""Benchmark: DataFrame row-wise transform on 10k-row DataFrame"""
import json, time
import pandas as pd

ROWS = 10_000
WARMUP = 3
ITERATIONS = 10
df = pd.DataFrame({"a": [i * 1.0 for i in range(ROWS)], "b": [i * 2.0 for i in range(ROWS)]})

for _ in range(WARMUP):
    df.apply(lambda row: pd.Series({"a": row["a"] * 2, "b": row["b"] + 1}), axis=1)

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.apply(lambda row: pd.Series({"a": row["a"] * 2, "b": row["b"] + 1}), axis=1)
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "dataframe_transform_rows", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))
